### ISSUE TITLE
Profiles missing in <tftpboot>/pxelinux.cfg/default

### ISSUE THREAD

**Issue by dywisor** [cobbler/cobbler#2196]
Using current git HEAD, the generated ``pxelinux.cfg/default`` lacks any entries from profiles.

In ``cobbler/tftpgen.py``:

``make_pxe_menu()`` calls ``get_menu_items()`` without specifying ``arch``, so it defaults to ``None``. 
``get_menu_items()`` then compares each profile's arch against ``None``, which never matches.
As a result, the ``profile_list`` is empty.

Introduced in commit be44b46e3b23c5a2791a02984933b5b85d82feef.

Proposed fix (partial revert of the commit referenced above):
```diff
diff --git a/cobbler/tftpgen.py b/cobbler/tftpgen.py
index 5ae5e7ac..488046b3 100644
--- a/cobbler/tftpgen.py
+++ b/cobbler/tftpgen.py
@@ -288,7 +288,8 @@ class TFTPGen(object):
         profile_list = [profile for profile in self.profiles]
         profile_list = sorted(profile_list, key=lambda profile: profile.name)

-        profile_list = [profile for profile in profile_list if profile.get_arch() == arch]
+        if arch:
+            profile_list = [profile for profile in profile_list if profile.get_arch() == arch]

         # sort the images
         image_list = [image for image in self.images]
```

**Comment by SchoolGuy** [comment]
@loreson This was your PR. Please see if this affects the functionallity you implemented and start a PR for the fix of the error described here!

**Comment by watologo1** [comment]
@loreson Yep, please have a look at this...

**Comment by watologo1** [comment]
On the other hand side: pxelinux/syslinux is deprecated and may vanish at some point of time (still the regression should get fixed on short term).
Better go for a grub2 solution.

Also see:
https://github.com/cobbler/cobbler/issues/1828

**Comment by SchoolGuy** [comment]
This was fixed by #2216
